docs/openapi-docs/src/test/scala/sttp/tapir/docs/openapi/VerifyYamlTest.scala

```diff
@@ -538,6 +538,15 @@ class VerifyYamlTest extends AnyFunSuite with Matchers {
     val actualYamlNoIndent = noIndentation(actualYaml)
     actualYamlNoIndent shouldBe expectedYaml
   }
+
+  test("explicit Content-Type header should have priority over the codec") {
+    val ep = endpoint.out(stringBody and header("Content-Type", "text/csv"))
```

Comment:
I think it also might be beneficial to 
1. move this definition to `tests/src/main/scala/sttp/tapir/tests/package.json` (a reusable endpoint definition)
2. add a server test (in `ServerBasicTests`) that when this endpoint is called, we get response with the correct content-type

Reply:
@adamw point 1. and 2. implemented, please review